src/test/java/com/artipie/maven/MavenITCase.java

```diff
@@ -0,0 +1,284 @@
+            )
+        );
+    }
+
+    @ParameterizedTest
+    @CsvSource({"helloworld,0.1,0.1", "snapshot,1.0-SNAPSHOT,1.0-[\\d-.]{17}"})
+    void deploysArtifact(final String type, final String vers, final String assembly)
+        throws Exception {
+        this.prepareDirectory(
+            String.format("%s-src", type),
+            String.format("%s-src/pom.xml", type)
+        );
+        MatcherAssert.assertThat(
+            "Build failure",
+            this.exec(
+                "mvn", "-s", "/home/settings.xml", "-f",
+                String.format("/home/%s-src/pom.xml", type),
+                "deploy"
+            ).replaceAll("\n", ""),
+            new StringContains("BUILD SUCCESS")
+        );
+        this.exec(
+            "mvn", "-s", "/home/settings.xml", "-f",
+            String.format("/home/%s-src/pom.xml", type),
+            "clean"
+        );
+        MatcherAssert.assertThat(
+            "Artifacts weren't added to storage",
+            this.storage.list(
+                new Key.From("repos", "my-maven", "com", "artipie", type)
+            ).join().stream()
+            .map(Key::string)
+            .collect(Collectors.toList())
+            .toString()
+            .replaceAll("\n", ""),
+            new AllOf<>(
+                Arrays.asList(
+                    new MatchesPattern(
+                        Pattern.compile(
+                            String.format(
+                                ".*repos/my-maven/com/artipie/%s/maven-metadata.xml.*", type
+                            )
+                        )
+                    ),
+                    new MatchesPattern(
+                        Pattern.compile(
+                            String.format(
+                                ".*repos/my-maven/com/artipie/%s/%s/%s-%s.pom.*",
+                                type, vers, type, assembly
+                            )
+                        )
+                    ),
+                    new MatchesPattern(
+                        Pattern.compile(
+                            String.format(
+                                ".*repos/my-maven/com/artipie/%s/%s/%s-%s.jar.*",
+                                type, vers, type, assembly
+                            )
+                        )
+                    )
+                )
+            )
+        );
+    }
+
+    @AfterEach
+    void release() {
+        this.server.stop();
+        this.cntn.stop();
+    }
+
+    private void prepareDirectory(final String src, final String pom) throws IOException {
+        FileUtils.copyDirectory(
+            new TestResource(src).asPath().toFile(),
+            this.subdir.resolve(src).toFile()
+        );
+        Files.write(
+            this.subdir.resolve(pom),
+            String.format(
+                Files.readString(this.subdir.resolve(pom)),
+                this.port
+            ).getBytes()
+        );
+    }
+
+    /**
+     * Executes dnf command in container.
```

Comment:
@genryxy javadoc seems inaccurate, I think we can remove it as it's not required in tests and other methods have no comments